### PR TITLE
Run acceptance tests with ~fake~ real browsers

### PR DESCRIPTION
**Pull request by gabriel-v**
This patch installs and uses Firefox (and Chrome, if I get it working) to test the following:
- initial wizard
- login into all apps

TODO:
- test things around the Admin UI
- do very simple tests in the apps
- also use Chrome to run all the tests

### CODE REVIEW

**Comment by gabriel-v** [comment]
So Chrome doesn't want to start for some reason:

        E       selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally
        E         (Driver info: chromedriver=2.33.506092 (733a02544d189eeb751fe0d7ddca79a0ee28cce4),platform=Linux 4.4.0-81-generic x86_64)

**Comment by gabriel-v** [comment]
We're suspecting that the wait time per page navigation is too short. Let's add them in the tests from here:

- https://splinter.readthedocs.io/en/latest/drivers/firefox.html#splinter.driver.webdriver.firefox.WebDriver
- https://splinter.readthedocs.io/en/latest/drivers/chrome.html#splinter.driver.webdriver.chrome.WebDriver

Also, we should add a log file and verbose parameter to the chrome driver, to debug it

File: tests/requirements.in [hunk 1]
```diff
@@ -1,3 +1,6 @@
 pytest
 requests
 pip-tools
+selenium
+splinter
+
```
Comment (mgax) [inline tests/requirements.in]: ?

File: tests/testsuite/test_integration_with_browsers.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
+import os
+import splinter
+import pytest
+
+DOMAIN = 'liquid.example.org'
+URL = 'http://'+DOMAIN
```
Comment (mgax) [inline tests/testsuite/test_integration_with_browsers.py]: py36: `f'{DOMAIN}'`, which means `'{DOMAIN}'.format(**locals())`

File: tests/testsuite/test_integration_with_browsers.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    assert browser.is_text_present("Users")
+    assert browser.is_text_present("Active users")
+    assert browser.is_text_present("Inactive users")
+
+    browser.click_link_by_href('/admin-ui/discovery')
+    assert browser.is_text_present("Discovery")
+    assert browser.is_text_present("Trusted nodes")
+    assert browser.is_text_present("Untrusted nodes")
+
+    browser.click_link_by_href('/admin-ui/about')
```
Comment (mgax) [inline tests/testsuite/test_integration_with_browsers.py]: `<3`

File: tests/testsuite/test_integration_with_browsers.py [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+    # let's log out from the django admin
+    browser.find_by_text('Log out').click()
+
+    # and check that we're logged out
+    browser.find_by_id('loggedin-btngroup').click()
+    assert browser.is_text_present("login")
+
+# no chrome on admin UI because clickin issues in menu
+@pytest.mark.parametrize('browser', ['firefox'], indirect=True)
```
Comment (mgax) [inline tests/testsuite/test_integration_with_browsers.py]: `indirect=True` `<3`

File: tests/testsuite/test_integration_with_browsers.py [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    'chrome': {
+        'service_args': ['--verbose', '--log-path=chromedriver.log'],
+    },
+}
+
+
+WELCOME_FILE_PATH = '/var/lib/liquid/core/welcome_done'
+
+def skip_if_welcome_not_set():
+    if not os.path.isfile(WELCOME_FILE_PATH):
```
Comment (mgax) [inline tests/testsuite/test_integration_with_browsers.py]: The concept is sound, but! you're testing the system as a black box over HTTP. You should check for welcome screen via HTTP, e.g. by doing a GET on `/` and checking if it redirects to `/welcome`.

File: tests/testsuite/test_integration_with_browsers.py [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+    assert browser.is_text_present("Upload key")
+
+    browser.click_link_by_href('/admin-ui/services')
+    assert browser.is_text_present("Services")
+    for app_name in APP_NAMES:
+        assert browser.is_text_present(app_name.upper())
+    for app_desc in ['Search Tool', 'Annotations', 'Chat', 'Wiki', 'File Sharing']:
+        assert browser.is_text_present(app_name.upper())
+
+    browser.click_link_by_href('/admin-ui/users')
```
Comment (mgax) [inline tests/testsuite/test_integration_with_browsers.py]: Even though it takes maybe 15s per browser instance, i think we should separate these into individual tests.

File: tests/requirements.in (hunk 1 above)
Comment (mgax) [reply tests/requirements.in]: final empty line. still there.

**Pull request merged**